build/template-OneBranch-CI-libsandsamples.yaml

```diff
@@ -21,27 +22,45 @@ steps:
   inputs:
     version: 6.x
 
-- task: DotNetCoreCLI@2
-  displayName: 'dotnet workload restore for Desktop project'
-  inputs:
-    command: 'custom'
-    custom: 'workload'
-    arguments: 'restore microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client.Desktop\Microsoft.Identity.Client.Desktop.csproj'
+- ${{ if eq(parameters.Mode, 'build') }}:
+  - task: DotNetCoreCLI@2
+    displayName: 'dotnet workload restore for Desktop project'
+    inputs:
+      command: 'custom'
+      custom: 'workload'
+      arguments: 'restore microsoft-authentication-library-for-dotnet\src\client\Microsoft.Identity.Client.Desktop\Microsoft.Identity.Client.Desktop.csproj'
```

Comment:
There must be an env variable that points to the root of the repository. Having to change paths like this in each task is quite ugly.

Reply:
Maybe build.sourcesdirectory ? https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml